test/functional/plugin/health_spec.lua

```diff
@@ -48,68 +49,144 @@ describe('health.vim', function()
     command("set runtimepath+=test/functional/fixtures")
   end)
 
-  it("health#report_*()", function()
-    helpers.source([[
-      let g:health_report = execute([
-        \ "call health#report_start('Check Bar')",
-        \ "call health#report_ok('Bar status')",
-        \ "call health#report_ok('Other Bar status')",
-        \ "call health#report_warn('Zub')",
-        \ "call health#report_start('Baz')",
-        \ "call health#report_warn('Zim', ['suggestion 1', 'suggestion 2'])"
-        \ ])
-    ]])
-    local result = helpers.eval("g:health_report")
-
-    helpers.eq(helpers.dedent([[
-
-
-      ## Check Bar
-        - OK: Bar status
-        - OK: Other Bar status
-        - WARNING: Zub
-
-      ## Baz
-        - WARNING: Zim
+  describe(":checkhealth", function()
+    it("functions health#report_*() render correctly", function()
```

Comment:
@justinmk went ahead and created this test case to render a checkhealth that uses all the function features (advice, help docs references) to compensate for the previous one